src/include/encoding.h

```diff
@@ -125,6 +125,18 @@ WRITE_INTTYPE_ENCODER(int32_t, le32)
 WRITE_INTTYPE_ENCODER(uint16_t, le16)
 WRITE_INTTYPE_ENCODER(int16_t, le16)
 
+#if defined(__APPLE__) && defined(__MACH__)
+inline void encode(const size_t &v, bufferlist& bl) {
```

Comment:
it is a distinct type on Darwin, which is why the linking fails to find the right overload, and here we provide one under the appropriate preprocessor conditions
I have considered refactoring the places where size_t is used as the input, or casting those to uint_64, but it ended up too many places and also didn't look good. 

I am working now to change this preprocessor conditional compilation into a c++ template with a constraint that size_t is a distinct type. I believe that will be a better solution because it won't be conditioned on platform or compiler

Reply:
> I have considered refactoring the places where size_t is used as the input, or casting those to uint_64, but it ended up too many places and also didn't look good.

Sigh...

> I am working now to change this preprocessor conditional compilation into a c++ template with a constraint that size_t is a distinct type. I believe that will be a better solution because it won't be conditioned on platform or compiler

+1

Implementation-wise, it would probably be clearer if you just invoke a `uint64_t` overload of `encode` rather than open-code the little-endian conversion.

Reply:
> Implementation-wise, it would probably be clearer if you just invoke a uint64_t overload of encode rather than open-code the little-endian conversion.

Originally I discarded that approach because it ended up being almost the same lines of code, but now that you mention it I agree it would be cleaner to call the standard implementation and only deal with the type conversion

Reply:
Please see the latest version (will upload soon if not yet there)